
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../intro/">
      
      
        <link rel="next" href="../assembly/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.11">
    
    
      
        <title>Quality Control - Biosb Computational Metagenomics</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.4af4bdda.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  
  
    
  
  
  <meta property="og:type" content="website" />
  <meta property="og:title" content="Biosb Computational Metagenomics - Quality Control" />
  <meta property="og:description" content="None" />
  <meta property="og:url" content="None" />
  <meta property="og:image" content="Noneassets/images/banner.png" />
  <meta property="og:image:type" content="image/png" />
  <meta property="og:image:width" content="1200" />
  <meta property="og:image:height" content="630" />

  <link rel="alternate" type="application/rss+xml" title="RSS feed of created content" href="Nonefeed_rss_created.xml">
  <link rel="alternate" type="application/rss+xml" title="RSS feed of updated content" href="Nonefeed_rss_updated.xml">

  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#fastq-format" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Biosb Computational Metagenomics" class="md-header__button md-logo" aria-label="Biosb Computational Metagenomics" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Biosb Computational Metagenomics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Quality Control
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/mdehollander/biosb-metagenomics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    biosb-metagenomics
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Biosb Computational Metagenomics" class="md-nav__button md-logo" aria-label="Biosb Computational Metagenomics" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    Biosb Computational Metagenomics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/mdehollander/biosb-metagenomics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    biosb-metagenomics
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 1: Data processing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Day 1: Data processing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../intro/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Quality Control
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Quality Control
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#fastq-format" class="md-nav__link">
    <span class="md-ellipsis">
      FastQ format
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fastqc-reports" class="md-nav__link">
    <span class="md-ellipsis">
      FastQC reports
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-trimming-and-filtering" class="md-nav__link">
    <span class="md-ellipsis">
      Quality trimming and filtering
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Quality trimming and filtering">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#alternative-trimming-with-bbduk" class="md-nav__link">
    <span class="md-ellipsis">
      Alternative trimming with bbduk
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#remove-phix-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      Remove PhiX sequences
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#from-one-sample-to-many" class="md-nav__link">
    <span class="md-ellipsis">
      From one sample to many
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quickly-check-what-is-in-this-metagenome" class="md-nav__link">
    <span class="md-ellipsis">
      Quickly check what is in this metagenome
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Quickly check what is in this metagenome">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sendsketch" class="md-nav__link">
    <span class="md-ellipsis">
      Sendsketch
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sourmash" class="md-nav__link">
    <span class="md-ellipsis">
      Sourmash
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kraken" class="md-nav__link">
    <span class="md-ellipsis">
      Kraken
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../assembly/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../binning/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Binning
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../cave-expedition/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cave expedition
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 3: Functional annotation
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Day 3: Functional annotation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../functional-annotation/viromes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Viromics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../functional-annotation/bgcs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BGCs
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#fastq-format" class="md-nav__link">
    <span class="md-ellipsis">
      FastQ format
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fastqc-reports" class="md-nav__link">
    <span class="md-ellipsis">
      FastQC reports
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-trimming-and-filtering" class="md-nav__link">
    <span class="md-ellipsis">
      Quality trimming and filtering
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Quality trimming and filtering">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#alternative-trimming-with-bbduk" class="md-nav__link">
    <span class="md-ellipsis">
      Alternative trimming with bbduk
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#remove-phix-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      Remove PhiX sequences
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#from-one-sample-to-many" class="md-nav__link">
    <span class="md-ellipsis">
      From one sample to many
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quickly-check-what-is-in-this-metagenome" class="md-nav__link">
    <span class="md-ellipsis">
      Quickly check what is in this metagenome
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Quickly check what is in this metagenome">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sendsketch" class="md-nav__link">
    <span class="md-ellipsis">
      Sendsketch
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sourmash" class="md-nav__link">
    <span class="md-ellipsis">
      Sourmash
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#kraken" class="md-nav__link">
    <span class="md-ellipsis">
      Kraken
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


  <h1>Quality Control</h1>

<div class="admonition important">
<p class="admonition-title">Learning objectives</p>
<ul>
<li>Understanding the FastQ format</li>
<li>Interpret FastQC reports</li>
<li>Create high quality reads by trimmign and filtering with FastP and BBduk</li>
</ul>
</div>
<h2 id="fastq-format">FastQ format<a class="headerlink" href="#fastq-format" title="Permanent link">&para;</a></h2>
<p>First let's take a look at the data</p>
<pre><code>zcat /data/reads/sample_0.fq.gz | head
</code></pre>
<details class="done">
<summary>The output looks like this</summary>
<pre><code>@gi|1184849861|gb|KY629563.1|-34525/1
CTGATCAACGTGGGAATGAACATGGAGCAGAGCGGCCAGTGTCTCAACCTGTGCTCCGCCTTCATGCACTCTCGCGGCGCCTTCAAGCCCGGCGACATCGACGTGGAAATTTTCACCATGCCCACCAAGTTCGGGCGCGTCAGCACCACG
+
DDDGGEG*DIIGH?KKHJGHGKKKJKKJJ9JDK=JKFKKGJHIKHEIJJJKKGIHKGDE1DKA&gt;KEGBEEEBGI?BBEGHEEEFEEE9ED;FCB3BEEDCBEE5EEC'CA$E9EEEAEEEFEE?FFCEEE;$EE)DBCDEEDDECAAE'$
@gi|1184849861|gb|KY629563.1|-34523/1
GCCTTTGTCGGGTAAGGGGTGTGGCCCTCCTCCCGACAAGGCGGGCCACGGTTCGCCAGCGAACTAGGTCGGGGAGCTGGGAAGGAGCCGGAATCGGGTGGCCCCAATTTCGGGGAGAGGTTTGGGCGTCAGCCGCCCGGAAGCTCGTCG
+
DDDE2GGGIIIIIKKJKHJKEDJ=AIHKKHKJKKKKJHBIE$KJJJCEKJB=JH@JKEHKGEEEEDAJECDGC?EIFEBEDDF6FGEEDEE$FBDCEEA@EE$4EE$E??CDE?ED;CCDE1DEBE;ECC$?$$AEDA;EDD@$EEDE=F
@gi|1184849861|gb|KY629563.1|-34521/1
AGACCGAGCCCTTCCTTATAGTGGATTTCTCCGGTTCCGTCAACCAAATTTGCAGTAATGCGGGAGCGACGCTTCCACGAAATGGTCACTGTCCCGTCCACCTCGGAGAGCTTTACGCTTTCCGGTGTGTAGGGCTTGAGATCGATCGAT
</code></pre>
</details>
<p>There is a pattern that is repeated every 4 lines. This is the <a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ format</a>. It follows this structure:</p>
<table>
<thead>
<tr>
<th>Line</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>Always begins with '@' and then information about the read</td>
</tr>
<tr>
<td>2</td>
<td>The actual DNA sequence</td>
</tr>
<tr>
<td>3</td>
<td>Always begins with a '+' and sometimes the same info in line 1</td>
</tr>
<tr>
<td>4</td>
<td>Has a string of characters which represent the quality scores; must have same number of characters as line 2</td>
</tr>
</tbody>
</table>
<p>The fourth line shows the quality of the read. To make the sequence and the qaility align well, the numerical score is converted into a code where each individual character represents the numerical quality of an individual nucleotide, following this scheme:</p>
<div class="admonition done">
<p class="admonition-title">FASTQ quality encoding</p>
<pre><code>    Quality encoding: !"#$%&amp;'()*+,-./0123456789:;&lt;=&gt;?@ABCDEFGHIJ
                    |         |         |         |         |
    Quality score:    01........11........21........31........41
</code></pre>
</div>
<p>NextSeq and NovaSeq data often contains poly-g tails. Let's check if this also is the case for our simulated input</p>
<pre><code>$ zcat /data/reads/sample_0.fq.gz | grep -E "GGGGGGGGGGGGGGGGGG$"
</code></pre>
<p>This search does not retun any hits. But often the end of the sequences contain stretches of poly-g. Below is an example how poly-g tails look like in Novaseq data.</p>
<details class="done">
<summary>Sequence data with poly-g tails</summary>
<pre><code>GGCCAGGACCACGCGGTGGAGCAGCGCGCCGCGGCGCCGGGCGCGCTTGACGACCAGTCTCTTATAAACATATCCCAGCCCACGAGACCACCAGTTGCATCTCGTATTCCGTCTTATGCTTGTATATTGGGGGGGGGGGGGGGGGGGGGGG
GAGTCGATCGAGGAGATGAAGCACGCGGAGAAGGTCATTCACCGCATCCTCTACTTCGATGCTGTCTCTTATACACATCCCGAGCCCACGAGACCACCTGTTGCATCTCGTATGCCGTCTTCTGCTTGAAAAGGGGGGGGGGGGGGGGGGG
GGCCCGTGCAGTTCGAGATCATCTCCGAGCCCACGAGACGACCGGGTGCTTGGCGGGAGCGGCGGGGTGGTTTTATCTTCGTGGTGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG
GGGCGGCAACGCCGACACCTACCAGCTACAGCAGACGGTGCGCCGCACCATCGGCCGCTGGGTCGGCGGCCGGCTGCGCCGCCGTCCCAAGATAATCCCGGTGGTGGGGGGGGGTGGTTGGGGGGGTTGTGGGGGGGGGGGGGGGGGGGGG
GGTAGGGCCGCTCGAGAAGCTCGCACAGCATGCGGCCGAATTCGCGGTACATGCATACGTTGACGTCGGCGGGGGGGGGGGGGGGGGGAGAGCACCGGGGGGCGCACAGGGCGGCCGGGTGGGGGTCGTGGTGGGGGGGGGGGGGGGGGGG
GGGGAGGAGGGAGACGCGCCCGCGGGCGTGCCCGCCGCCGCGGCGATGCCCTTGAAGGCCGCTGGTGTGTCCTTCAGCGCCTGCGCCGCGAGTTCGCCGAACTGCTGCGTCAACGCGCCCCACCACTGCTGCGGGGGGGGGGGGGGGGGGG
GTACCCGAGCCGCTCCGCGTGCCGCCGGACCTCCGACGCGTGAGGCCCGGAGCCGGTCGCTGACCAGAGGGCCAGGCCGAAGCGATGGGGAGTGGTCGCGACGAACCCGCAGCGCTGTCCGGCGGCGGGGGGGGGGGGGGGGGGGGGGGGG
GAAGGTGTGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGTGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGCGGGGGGGGGGGGGGGCGGGGGCGGGGGGGGGGGGGGGGGGGGGGGGGGGG
GCGCCACGCCGAGCACCGACGGCATCATCGGCACCCACACGCCGTGTGTGAACCTGTCTCTTATACACATCTCCGAGCCACGAGACCACCTGTTGCATCTCGTATGCCGTCTTCTGCTTGAAAATGGGGGGGGGGGGGGGGGGGGGGGGGG
CTCTTCATCCGTTCCGGCGCCTGCATCCATTCCCGCGGCGCCGGTTGGCGGGGGGGGGGGGGGGGGGGGGGGGGGTGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG
</code></pre>
</details>
<h2 id="fastqc-reports">FastQC reports<a class="headerlink" href="#fastqc-reports" title="Permanent link">&para;</a></h2>
<p>Let's check the quality of the data with FastQC before quality filtering:</p>
<pre><code>mkdir fastqc_untrimmed_reads
fastqc --threads 8 /data/reads/sample_0.fq.gz -o fastqc_untrimmed_reads/
</code></pre>
<div class="admonition question exercise">
<p class="admonition-title">Question</p>
<p>Open the FastqQC report <code>fastqc_untrimmed_reads//sample_0_fastqc.html</code></p>
</div>
<h2 id="quality-trimming-and-filtering">Quality trimming and filtering<a class="headerlink" href="#quality-trimming-and-filtering" title="Permanent link">&para;</a></h2>
<p>Because NovaSeq and NextSeq from Illumina contain often poly-g tails, it is good to use a trimming tool that can detect poly-g tails. Fastp can do that. We keep the defaults like they are, but specify we have interleaved input data. In case you have R1 and R2 file for each sample, you need to enable adapter detection with the <code>--detect_adapter_for_pe</code> flag. Execute fastp with this command:</p>
<pre><code>fastp -i /data/reads/sample_0.fq.gz \
      --stdout \
      --interleaved_in \
      -q 25 \
      --cut_front \
      --cut_tail \
      --cut_mean_quality 25 \
      -l 51 \
      --thread 16 \
      --trim_poly_g &gt; sample_0.trim.fastq
</code></pre>
<p>Now make a FastQC report again, to see the results of the quality filtering.</p>
<pre><code>mkdir fastqc_trimmed_reads
fastqc --threads 8 sample_0.trim.fastq -o fastqc_trimmed_reads/
</code></pre>
<div class="admonition question">
<p class="admonition-title">Exercise</p>
<ul>
<li>Are there any adapter sequences detected?</li>
<li>Take a look at the html report</li>
</ul>
</div>
<h3 id="alternative-trimming-with-bbduk">Alternative trimming with bbduk<a class="headerlink" href="#alternative-trimming-with-bbduk" title="Permanent link">&para;</a></h3>
<p>Alternative trimming with bbduk. Compare poly-g tail filtering, adapter trimming.</p>
<pre><code>bbduk.sh in=/data/reads/sample_0.fq.gz  \
    out=sample_0.trim.bbduk.fastq.gz \
    interleaved=true \
    trimpolygright=1 \
    qtrim=w trimq=20 \
    minlength=51 \
    ref=/data/databases/nextera.fa.gz ktrim=r \
    stats=bbduk.stats \
    t=16
</code></pre>
<div class="admonition question exercise">
<p class="admonition-title">Question</p>
<ul>
<li>Create also an FastQC report for the trimming with bbduk </li>
<li>What are differences between filtering with fastp and bbduk?</li>
</ul>
</div>
<h2 id="remove-phix-sequences">Remove PhiX sequences<a class="headerlink" href="#remove-phix-sequences" title="Permanent link">&para;</a></h2>
<pre><code>bbmap.sh ref=/data/databases/phix174_ill.ref.fa.gz \
         in=sample_0.trim.fastq \
         interleaved=true \
         outu=sample_0.nophix.fastq.gz \
         outm=sample_0.phix.fastq.gz \
         t=4
</code></pre>
<div class="admonition question">
<p class="admonition-title">Exercise</p>
<ul>
<li>How many PhiX sequences are detected?</li>
<li>From which samples?</li>
<li>Can you confirm the sequences are PhiX?</li>
<li>The input data was simulated without adding any PhiX. How could there still PhiX sequences being detected?</li>
</ul>
</div>
<h2 id="from-one-sample-to-many">From one sample to many<a class="headerlink" href="#from-one-sample-to-many" title="Permanent link">&para;</a></h2>
<div class="admonition question">
<p class="admonition-title">From one sample to many</p>
<p>Now do a QC for all samples. You can use a for loop for that. For example, fastp can be run like:</p>
<pre><code>for file in /data/reads/*.gz; do \
    sample=$(basename ${file} .fq.gz);
    fastp -i $file --stdout --interleaved_in -q 25 --cut_front --cut_tail --cut_mean_quality 25 -l 51 --thread 16 --trim_poly_g &gt; $sample.trim.fastq;
    bbmap.sh ref=/data/databases/phix174_ill.ref.fa.gz in=$sample.trim.fastq interleaved=true outu=$sample.nophix.fastq.gz outm=$sample.phix.fastq.gz t=4;
done
</code></pre>
</div>
<h2 id="quickly-check-what-is-in-this-metagenome">Quickly check what is in this metagenome<a class="headerlink" href="#quickly-check-what-is-in-this-metagenome" title="Permanent link">&para;</a></h2>
<h3 id="sendsketch">Sendsketch<a class="headerlink" href="#sendsketch" title="Permanent link">&para;</a></h3>
<pre><code>sendsketch.sh --in=sample_0.nophix.fastq.gz threads=4 address=refseq
</code></pre>
<h3 id="sourmash">Sourmash<a class="headerlink" href="#sourmash" title="Permanent link">&para;</a></h3>
<p><strong>Note</strong>: The sourmash database is not included in the VM because it can't be downloaded at the moment. There is an <a href="https://sourmash.readthedocs.io/en/latest/databases.html">overview of all prepared databases</a> The <a href="https://farm.cse.ucdavis.edu/~ctbrown/sourmash-db/gtdb-rs207/gtdb-rs207.genomic-reps.dna.k51.lca.json.gz">51 kmer set of representative genomes</a> would be a good one to use when available</p>
<p>Now try it locally, using sourmash. First create a signature for a sigle sample.</p>
<pre><code>sourmash sketch dna -p scaled=10000,k=51 sample_0.nophix.fastq.gz -o sample_0.sig
</code></pre>
<p>Find what is in this metagenome using the <code>gather</code> command:</p>
<pre><code>sourmash gather sample_0.sig /data/databases/gtdb-rs207.genomic-reps.dna.k51.lca.json.gz
</code></pre>
<h3 id="kraken">Kraken<a class="headerlink" href="#kraken" title="Permanent link">&para;</a></h3>
<p>Compare with kraken</p>
<p>First setup the database</p>
<pre><code>mkdir kraken_db
tar zxvf /data/databases/k2_standard_08gb_20220926.tar.gz -C kraken_db
</code></pre>
<p>Now run kraken2</p>
<pre><code>kraken2 --db kraken_db/ --threads 16 --output sample_0.kraken --report sample_0.kraken.report --gzip-compressed --use-names sample_0.nophix.fastq.gz
</code></pre>
<div class="admonition question">
<p class="admonition-title">Question</p>
<p>Take a look at the <code>sample_0.kraken.report</code>. What are the differences in classification compared to sendsketch and sourmash?</p>
</div>







  
    
  
  


  <aside class="md-source-file">
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Last update">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"/></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date" title="April 2, 2025 14:38:26">April 2, 2025</span>
  </span>

    
    
    
      
  
  <span class="md-source-file__fact">
    <span class="md-icon" title="Contributors">
      
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 4a4 4 0 0 1 4 4 4 4 0 0 1-4 4 4 4 0 0 1-4-4 4 4 0 0 1 4-4m0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4"/></svg>
      
    </span>
    <nav>
      
        <a href="mailto:m.dehollander@nioo.knaw.nl">Mattias de Hollander</a>
    </nav>
  </span>

    
    
  </aside>





                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.expand"], "search": "../../assets/javascripts/workers/search.f8cc74c7.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
  
      <script src="../../assets/javascripts/bundle.c8b220af.min.js"></script>
      
    

  </body>
</html>